### PR TITLE
Implemented Checkboxes Multi/Single-Select Field

### PR DESCRIPTION
**Pull request by miguelaenlle**
https://github.com/user-attachments/assets/17410aae-4d1f-4926-b596-f7cbb1576b91

- Implemented `Checkboxes` component with multi-select and single-select capabilities, based on the `Checkboxes` component used in the HackIllinois 2024 Site

- Checkbox component is responsive to different screen sizes (though not mobile-optimized just yet):

Screenshot 2024-10-25 at 11 06 57 PM

- Checkbox component implements `useFormContext`, so it functions as any other form component

- Checkbox component lets users type in "Other" responses:

image

- Created temporary `TestForm` component to test the `Checkboxes` component with multiple questions, screen sizes, and to test the `useFormContext` functionality.

### CODE REVIEW

File: components/Form/Checkboxes/Checkboxes.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    label: string | number;
+    value: string | number;
+
+    // `isRadio: true` indicates that this specific checkbox will act like a radio button
+    // (i.e. if this is checked, then no other checkbox may be also checked)
+    isRadio?: boolean;
+
+    // `isRadioButton: true` indicates that this specific checkbox is a radio button
+    // and the css will change accordingly
+    isRadioButton?: boolean;
```
Comment (J164) [inline components/Form/Checkboxes/Checkboxes.tsx]: What is the difference between isRadio and isRadioButton?

File: components/Form/Checkboxes/Checkboxes.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    // there can be exactly one checkbox that serves as an "Other" option, and will allow the user
+    // to enter their own choice for the `value` through a text input
+    isOther?: boolean;
+    otherPlaceholder?: string;
```
Comment (J164) [inline components/Form/Checkboxes/Checkboxes.tsx]: Could we adjust this type so the placeholder can only be specified if isOther is true?

**Review by J164 — CHANGES_REQUESTED** [review]
This is a great start! Two questions: 
How can we make a CheckBox option required?
Can we make it so the Other text field is required when Other is selected?

File: components/Form/Checkboxes/Checkboxes.tsx (hunk 1 above)
Comment (miguelaenlle) [reply components/Form/Checkboxes/Checkboxes.tsx]: There's no difference -- this was part of the original Checkboxes component from site-2024. I'll merge this and isRadio

File: components/Form/Checkboxes/Checkboxes.tsx (hunk 2 above)
Comment (miguelaenlle) [reply components/Form/Checkboxes/Checkboxes.tsx]: I'll add that

**Comment by miguelaenlle** [comment]
> This is a great start! Two questions: 
> How can we make a CheckBox option required? 
> Can we make it so the Other text field is required when Other is selected?

https://github.com/user-attachments/assets/13d8c486-a18f-4aa2-9dc0-11aaa70d1c04

I added these functionalities in the last commit:
- Ability to make CheckBox required
- Other textfield is required when Other is selected (this is true for both if the whole CheckBox component is required or not)

- Removed isRadioButton from CheckboxOption Type
- Adjusted CheckboxOption Type such that otherPlaceholder can only be specified if isOther is true

**Review by J164 — APPROVED** [review]
Looks good! Thanks for the changes!

**Pull request merged**